applications/ParticleMechanicsApplication/custom_conditions/particle_based_conditions/mpm_particle_lagrange_dirichlet_condition.cpp

```diff
@@ -0,0 +1,458 @@
+
+        KRATOS_CHECK_DOF_IN_NODE(VECTOR_LAGRANGE_MULTIPLIER_X,rnode)
+        KRATOS_CHECK_DOF_IN_NODE(VECTOR_LAGRANGE_MULTIPLIER_Y,rnode)
+        KRATOS_CHECK_DOF_IN_NODE(VECTOR_LAGRANGE_MULTIPLIER_Z,rnode)
+    }
+
+    return 0;
+}
+
+void MPMParticleLagrangeDirichletCondition::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo )
+{
+    KRATOS_TRY
+
+    GeometryType& r_geometry = GetGeometry();
+    const unsigned int number_of_nodes = r_geometry.size();
+    const unsigned int dimension = r_geometry.WorkingSpaceDimension();
+    if (rResult.size() != dimension * number_of_nodes * 2)
+    {
+        rResult.resize(dimension * number_of_nodes * 2 ,false);
+    }
+
+
+    if(dimension == 2)
+    {
+        for (unsigned int i = 0; i < number_of_nodes; ++i)
+        {
+            unsigned int index = i * dimension;
+            rResult[index    ] = r_geometry[i].GetDof(DISPLACEMENT_X).EquationId();
+            rResult[index + 1] = r_geometry[i].GetDof(DISPLACEMENT_Y).EquationId();
+
+
+            index = i * dimension + number_of_nodes * dimension;
+            rResult[index    ] = r_geometry[i].GetDof(VECTOR_LAGRANGE_MULTIPLIER_X).EquationId();
+            rResult[index + 1] = r_geometry[i].GetDof(VECTOR_LAGRANGE_MULTIPLIER_Y).EquationId();
+        }
+    }
+    else
+    {
+        for (unsigned int i = 0; i < number_of_nodes; ++i)
+        {
+            unsigned int index = i * dimension;
+            rResult[index    ] = r_geometry[i].GetDof(DISPLACEMENT_X).EquationId();
+            rResult[index + 1] = r_geometry[i].GetDof(DISPLACEMENT_Y).EquationId();
+            rResult[index + 2] = r_geometry[i].GetDof(DISPLACEMENT_Z).EquationId();
+
+            index = i * dimension + number_of_nodes * dimension;
+            rResult[index    ] = r_geometry[i].GetDof(VECTOR_LAGRANGE_MULTIPLIER_X).EquationId();
+            rResult[index + 1] = r_geometry[i].GetDof(VECTOR_LAGRANGE_MULTIPLIER_Y).EquationId();
+            rResult[index + 2] = r_geometry[i].GetDof(VECTOR_LAGRANGE_MULTIPLIER_Z).EquationId();
+        }
+    }
+
+
+    KRATOS_CATCH("")
+}
+
+void MPMParticleLagrangeDirichletCondition::GetDofList(
+    DofsVectorType& rElementalDofList,
+    ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY
+
+    GeometryType& r_geometry = GetGeometry();
+    const unsigned int number_of_nodes = r_geometry.size();
+    const unsigned int dimension =  r_geometry.WorkingSpaceDimension();
+    rElementalDofList.resize(0);
+    rElementalDofList.reserve(dimension * number_of_nodes *2);
+
+    GeneralVariables Variables;
+    const array_1d<double,3> & xg_c = this->GetValue(MPC_COORD);
+    // Calculating shape function
+    Variables.N = this->MPMShapeFunctionPointValues(Variables.N, xg_c);
+    if(dimension == 2)
+    {
+        for (unsigned int i = 0; i < number_of_nodes; ++i)
+        {
+            rElementalDofList.push_back( r_geometry[i].pGetDof(DISPLACEMENT_X));
```

Comment:
I am quite sure the code can be simplified, look how is done in other elements and conditions, idem for the previous class